src/AffectedUserFlow/PublishConsent/PublishConsentForm.tsx

```diff
@@ -65,19 +122,13 @@ const PublishConsentForm: FunctionComponent<PublishConsentFormProps> = ({
       revisionToken,
     )
     setIsLoading(false)
-    if (response.kind === "success") {
-      storeRevisionToken(response.body.revisionToken)
+    if ((response as PostKeysSuccess).revisionToken) {
```

Comment:
this casting jumps out as smell, should we introduce an intermediate representation for the possible responses?

Reply:
What kind of smell? HMW avoid this typecasting on a union type?

Reply:
Do you mean like having a distinction amongst the kinds of responses like the type argument on the original Network Response?

Reply:
@johnschoeman I added a kind attribute to avoid type casting, IMO these are pretty similar, but if it improves readability I'm all for it